File: C++/OOP/Inheritance/Multiple inheritance (example 1).cpp
#include <iostream>
using namespace std;

class Institute {
    public:
        string institute_name = "Institute of Engineering"; //just a random name
};
class Teacher {
    public:
        string teacher_name = "Bjarne Stroustrup";
        string subject = "C Plus Plus";
};
class Student {
    public:
        string student_name = "Abdullah Al Naiem";
};

class MultipleInheritance : public Institute, public Teacher, public Student {
    public:
        void displayInfo(){
            cout << institute_name << endl;
            cout << "Teacher: " << teacher_name << endl;
            cout << "Subject: " << subject << endl;
            cout << "Student: " << student_name << endl;
        }
};

int main(){
    MultipleInheritance obj;
    obj.displayInfo();
}

File: Python/OOP/Inheritance/Multiple inheritance (example 1).py
class Institute:
    institute_name = "Institute of Engineering"  # just a random name


class Teacher:
    teacher_name = "Guido van Rossum"
    subject = "Python Programming"


class Student:
    student_name = "Abdullah Al Naiem"


# inherit multiple classes at once and combind into a method
class MultipleInheritance(Institute, Teacher, Student):
    def displayInfo(self):
        print(self.institute_name)
        print("Teacher: ", self.teacher_name)
        print("Subject: ", self.subject)
        print("Student: ", self.student_name)


obj = MultipleInheritance()
obj.displayInfo()